test/StripNewlinesTest.php

```diff
@@ -4,82 +4,37 @@
 
 namespace LaminasTest\Filter;
 
-use Laminas\Filter\StripNewlines as StripNewlinesFilter;
+use Laminas\Filter\StripNewlines;
+use LaminasTest\Filter\TestAsset\StringableObject;
 use PHPUnit\Framework\Attributes\DataProvider;
 use PHPUnit\Framework\TestCase;
-use stdClass;
-
-use function array_keys;
-use function array_values;
 
 class StripNewlinesTest extends TestCase
 {
-    /**
-     * Ensures that the filter follows expected behavior
-     */
-    public function testBasic(): void
-    {
-        $filter         = new StripNewlinesFilter();
-        $valuesExpected = [
-            ''                                    => '',
-            "\n"                                  => '',
-            "\r"                                  => '',
-            "\r\n"                                => '',
-            '\n'                                  => '\n',
-            '\r'                                  => '\r',
-            '\r\n'                                => '\r\n',
-            "Some text\nthat we have\r\nstuff in" => 'Some textthat we havestuff in',
-        ];
-        foreach ($valuesExpected as $input => $output) {
-            self::assertSame($output, $filter($input));
-        }
-    }
-
-    public function testArrayValues(): void
-    {
-        $filter   = new StripNewlinesFilter();
-        $expected = [
-            "Some text\nthat we have\r\nstuff in" => 'Some textthat we havestuff in',
-            "Some text\n"                         => 'Some text',
-        ];
-        self::assertSame(array_values($expected), $filter(array_keys($expected)));
-    }
-
-    /** @return list<array{0: mixed}> */
-    public static function returnUnfilteredDataProvider(): array
+    /** @return array<string, array{0: mixed, 1: mixed}> */
+    public static function basicDataProvider(): array
     {
         return [
-            [null],
-            [new stdClass()],
+            'Empty String'        => ['', ''],
+            'Null'                => [null, null],
+            'String without CRLF' => ['string', 'string'],
+            'String with CRLF'    => ["Some\nText\r\nHere", 'SomeTextHere'],
+            'Escaped CRLF'        => ['\r\n', '\r\n'],
+            'Stringable'          => [new StringableObject("Hey\nThere"), 'HeyThere'],
+            'Integer'             => [1, '1'],
+            'Float'               => [1.23, '1.23'],
+            'True'                => [true, '1'],
+            'False'               => [false, ''],
+            'Array'               => [['a' => "Hey\rThere", 'b' => null], ['a' => 'HeyThere', 'b' => null]],
+            'Nested Array'        => [['a' => ["Hey\rThere"]], ['a' => ['HeyThere']]],
```

Comment:
Interesting! But this is not documented: https://docs.laminas.dev/laminas-filter/v2/standard-filters/#stripnewlines